core/modules/loaders.rs

```diff
@@ -26,17 +29,19 @@ pub trait ModuleLoader {
   /// algorithm described here:
   /// <https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier>
   ///
-  /// `is_main` can be used to resolve from current working directory or
+  /// [`ResolutionKind::MainModule`] can be used to resolve from current working directory or
   /// apply import map for child imports.
   ///
-  /// `is_dyn_import` can be used to check permissions or deny
+  /// [`ResolutionKind::DynamicImport`] can be used to check permissions or deny
   /// dynamic imports altogether.
   fn resolve(
     &self,
     specifier: &str,
     referrer: &str,
-    kind: ResolutionKind,
-  ) -> Result<ModuleSpecifier, Error>;
+    _kind: ResolutionKind,
+  ) -> Result<ModuleSpecifier, Error> {
+    Ok(resolve_import(specifier, referrer)?)
+  }
```

Comment:
I'm not a fan of this change - it's up to implementer to decide on the resolve implementation. Adding a default implementation might be a footgun.

Reply:
I can revert this part -- many of the implementations outside of deno are just calling `resolve_import`, however.

I suspect it might be better for us to split resolution and loading at some point in the future/

Reply:
Maybe, let's keep it reverted for now please